src/app/TargetView/TargetContextSelector.tsx

```diff
@@ -49,20 +50,23 @@ export const TargetContextSelector: React.FC<TargetContextSelectorProps> = ({ cl
   const [targets, setTargets] = React.useState<Target[]>([]);
   const [selectedTarget, setSelectedTarget] = React.useState<Target>();
   const [favorites, setFavorites] = React.useState<string[]>(getFromLocalStorage('TARGET_FAVORITES', []));
+  const [searchTerm, setSearchTerm] = React.useState<string>('');
   const [isTargetOpen, setIsTargetOpen] = React.useState(false);
   const [isLoading, setLoading] = React.useState(false);
 
-  const handleSelectToggle = React.useCallback(() => setIsTargetOpen((old) => !old), [setIsTargetOpen]);
+  const onToggleClick = React.useCallback(() => {
+    setIsTargetOpen((v) => !v);
+  }, [setIsTargetOpen]);
 
-  const handleTargetSelect = React.useCallback(
-    (_, { target }, isPlaceholder) => {
+  const onSelect = React.useCallback(
+    (_, target) => {
       setIsTargetOpen(false);
-      const toSelect: Target = isPlaceholder ? undefined : target;
-      if (!isEqualTarget(toSelect, selectedTarget)) {
-        context.target.setTarget(toSelect);
+      if (!isEqualTarget(target, selectedTarget)) {
+        setSelectedTarget(target);
+        context.target.setTarget(target);
       }
     },
-    [setIsTargetOpen, selectedTarget, context.target],
+    [targets, setSelectedTarget, setIsTargetOpen, context.target],
```

Comment:
Some more eslint fixes?\

```
/home/thvo/workspace/cryostat-web/src/app/TargetView/TargetContextSelector.tsx
   69:5  error  React Hook React.useCallback has a missing dependency: 'selectedTarget'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  194:5  error  React Hook React.useCallback has an unnecessary dependency: 'favorites'. Either exclude it or remove the dependency array  react-hooks/exhaustive-deps
  224:5  error  React Hook React.useMemo has a missing dependency: 'onClearSelection'. Either include it or remove the dependency array    react-hooks/exhaustive-deps
```

Reply:
We might need to add a `pf5` into the set of branch that lints/format checks are running?

Reply:
Probably worthwhile. I don't think it's worthwhile for every upstream feature branch, but this is a pretty significant one in terms of time and surface area, so it'll make review nicer.

Reply:
Yeh, I thinking it could be a temporary addition, which can be removed in `pf5` -> `main` PR.